src/main/java/org/jenkinsci/plugins/durabletask/Controller.java

```diff
@@ -48,15 +53,43 @@
     /**
      * Checks whether the task has finished.
      * @param workspace the workspace in use
+     * @param launcher a way to start processes
      * @return an exit code (zero is successful), or null if the task appears to still be running
      */
-    public abstract @CheckForNull Integer exitStatus(FilePath workspace) throws IOException, InterruptedException;
+    public @CheckForNull Integer exitStatus(FilePath workspace, Launcher launcher) throws IOException, InterruptedException {
+        if (Util.isOverridden(Controller.class, getClass(), "exitStatus", FilePath.class)) {
+            return exitStatus(workspace);
+        } else {
+            throw new AbstractMethodError("implement exitStatus(FilePath, Launcher)");
+        }
+    }
+
+    /** @deprecated use {@link #exitStatus(FilePath, Launcher)} instead */
+    public @CheckForNull Integer exitStatus(FilePath workspace) throws IOException, InterruptedException {
+        return exitStatus(workspace, createLauncher(workspace));
+    }
 
     /**
      * Tries to stop any running task.
      * @param workspace the workspace in use
+     * @param launcher a way to start processes
      */
-    public abstract void stop(FilePath workspace) throws IOException, InterruptedException;
+    public void stop(FilePath workspace, Launcher launcher) throws IOException, InterruptedException {
+        if (Util.isOverridden(Controller.class, getClass(), "stop", FilePath.class)) {
+            stop(workspace);
+        } else {
+            throw new AbstractMethodError("implement stop(FilePath, Launcher)");
+        }
```

Comment:
Yeah makes me nervous too, especially since we lack a proper test framework for verifying binary compatibility.